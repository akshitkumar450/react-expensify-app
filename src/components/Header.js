import React from 'react';
import { Link, NavLink } from 'react-router-dom';
import { startLogout } from '../actions/auth'
import { connect } from 'react-redux'


// if we want to show a component on all the pages we put it outside the switch tag
const Header = (props) => (
    <div>
        <h1>Expensify App</h1>
        <NavLink to='/dashboard' activeClassName='active'>Home</NavLink>
        <NavLink to='/create' activeClassName='active'>Create</NavLink>
        <NavLink to='/edit' activeClassName='active'>edit</NavLink>

        <button onClick={props.dispatch(startLogout)}>logout</button>
    </div>
)

// const mapDispatchToProps = (dispatch) => {
//     return {
//         startLogout: () => dispatch(startLogout())
//     }
// }

export default connect()(Header)