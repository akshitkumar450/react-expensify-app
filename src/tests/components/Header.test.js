import React from 'react'
import Header from '../../components/Header'
import { shallow } from 'enzyme'

// if we use  snapshotSerializers in jest.config.json file  then toJSON will be used automatically
// import toJSON from 'enzyme-to-json'

// shalllow rendering ->it renders only the component
// full dom rendering-> it renders child compoenent also

// test('should render header correctly', () => {
//     const renderer = new ReactShallowRenderer()
//     renderer.render(<Header />)
//     //  for the first time it will create a new snapshot and it will pass the test case
//     // after the first test case it will match the output with the existing snapshot
//     expect(renderer.getRenderOutput()).toMatchSnapshot()
//     console.log(renderer.getRenderOutput());
// })

test('should render header correctly', () => {
    const wrapper = shallow(<Header />)
    // expect(toJSON(wrapper)).toMatchSnapshot()
    expect(wrapper).toMatchSnapshot()

    // expect(wrapper.find('h1').text()).toBe('Expensify App')
})