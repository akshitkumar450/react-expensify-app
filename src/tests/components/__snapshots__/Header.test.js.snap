// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render header correctly 1`] = `
<div>
  <h1>
    Expensify App
  </h1>
  <NavLink
    activeClassName="active"
    exact={true}
    to="/"
  >
    Home
  </NavLink>
  <NavLink
    activeClassName="active"
    to="/create"
  >
    Create
  </NavLink>
  <NavLink
    activeClassName="active"
    to="edit"
  >
    edit
  </NavLink>
</div>
`;
